### ISSUE TITLE
trace/dns: Include resolved IPs in the response event

### ISSUE THREAD

**Issue by mauriciovasquezbernal** [inspektor-gadget/inspektor-gadget#1415]
The trace dns gadget doesn't provide the IPs that are returned by the server in the response. This issue is about extending this gadget to provide this information.

**Comment by wedaly** [comment]
I started this a while back and then got busy with work stuff :) WIP in branch here: https://github.com/inspektor-gadget/inspektor-gadget/compare/main...wedaly:inspektor-gadget:trace-dns-ip

The main issue I ran into was a BPF verifier error until I upgraded the clang version. Discussed in Slack: https://kubernetes.slack.com/archives/CSYL75LF6/p1674487191154789

> I'm working on adding IP addresses from a DNS reply to the trace DNS gadget, and I'm seeing some surprising (to me) behavior from the eBPF verifier.
> The branch [trace-dns-ip-verifier-repro](https://github.com/inspektor-gadget/inspektor-gadget/compare/main...wedaly:inspektor-gadget:trace-dns-ip-verifier-repro) adds the code to loop through DNS answers and copy the IP addresses to the BPF event. There's a limit MAX_ADDR_ANSWERS of 4.
When I compile the eBPF objects using TARGET=amd64 go generate ./... it works as expected: I can see IP addresses from DNS replies when running trace dns -o json This is on Fedora 37, so the kernel is 6.1.6-200.fc37.x86_64 and the llvm/clang version is 15.0.7
When I compile the eBPF objects using make ebpf-objects , then I get a verifier error loading the program at instruction 169  ("invalid write to stack R3 off=4 size=2"), which is the __builtin_memset call on line 136 in "dns.c". The full output from the verifier is attached below. This uses the ebpf-builder image, which is using llvm/clang version 11.
If I set MAX_ADDR_ANSWERS to 1, then the verifier accepts the program. I suspect in that case clang might be optimizing out something that's triggering the verification error.
I've tried commenting out parts of the loop in load_addresses to narrow down the issue, but I've been unable to get the program to pass verification. If I comment out references to the event pointer, I start getting verification errors about accessing the sk_buff data.
